presto-docs/src/main/sphinx/admin/resource-groups.rst

```diff
@@ -9,287 +9,452 @@ Except for the limit on queued queries, when a resource group runs out of a reso
+number of queries. This is particularly useful in multi-tenant environments where you want to ensure that no single user
+or query monopolizes the system resources.
+
+Resource Consumption
+====================
+
+A query belongs to a single resource group, and it consumes resources from that group,
+as well as its parent groups. If a resource group runs out of a certain resource, it does not cause running
+queries to fail. Instead, new queries will be queued until resources become available again.
+
+Sub-Groups and Query Acceptance
+===============================
+
+A resource group can either have sub-groups or accept queries, but it cannot do both.
+Sub-groups allow for hierarchical resource allocation, where each sub-group can have its own resource limits and
+queueing policies. On the other hand, a resource group that accepts queries directly is a leaf group, and it
+executes queries using its allocated resources.
+
+The resource groups and associated selection rules are configured by a manager, which is pluggable.
+
+Presto resource management can be done in two ways: File-based and Database-based resource group manager
+
+1. Add a file ``etc/resource-groups.properties``
+2. Set the ``resource-groups.configuration-manager`` property to ``file`` or ``db``
+3. Add further configuration properties for the desired manager.
+
+File-Based Resource Management
+------------------------------
+
+In a file-based resource manager, configuration information about the resource groups is stored in a JSON file.
+This file contains definitions of all resource groups and the rules to select them.
+The configuration file is loaded and used at the start of the Presto server.
+Any changes to the file require a restart of the Presto server to take effect.
+
+Database-Based Resource Management
+----------------------------------
+
+In a database-based resource manager, configuration information about the resource groups is stored in a relational
+database. The database contains definitions of all resource groups and the rules to select them.
+Unlike the file-based resource manager, changes to the configuration in the database take effect
+immediately and do not require a restart of the Presto server.
+
+Setting Up File-Based Resource Management
+-----------------------------------------
+
+Create a JSON file with the definitions of your resource groups and rules.
+Specify the location of this file in the Presto server's configuration file
+under the property ``resource-groups.config-file``.
+
+
+Setting Up Database-Based Resource Management
+---------------------------------------------
+
+Set up a relational database that Presto can access.
+Create tables in the database for resource groups and selection rules.
+Specify the database's JDBC URL in the Presto server's configuration file
+under the property ``resource-groups.config-db-url``.
+
+Both methods have their pros and cons. File-based resource management is simpler to set up but less flexible,
```

Comment:
I suggest moving this paragraph as a new second paragraph in Database-Based Resource Management. This explains the pros and cons of the two methods, and should be before the reader commits to choosing one in the Setting Up... sections.